app/presenters/search_presenters.py

```diff
@@ -29,33 +30,58 @@ def get_filters_from_default_question(question):
         }
 
     @staticmethod
-    def get_filters_from_question_with_options(question):
+    def get_filters_from_question_with_options(question, index):
         filter_group = []
-        for index, option in enumerate(question['options']):
-            filter_group.append({
+
+        if re.match('^serviceTypes(SCS|SaaS|IaaS)', question['id']):
+            filter_name = 'serviceTypes'
+        else:
+            filter_name = question['id']
+        for option in question['options']:
+            filter = {
                 'label': option['label'],
-                'name': question['id'],
-                'id': question['id'] + option['label'].lower(),
+                'name': filter_name,
+                'id': '%s-%s' % (filter_name, option['label'].lower()),
```

Comment:
`option['label']` can have spaces in it. We should replace them with something else.